File: e-commerce-backend/Models/Frontend/AccountModel.cs
using e_commerce_backend.Models.Backend;

namespace e_commerce_backend.Models.Frontend
{
    public class AccountModel
    {
        public Guid Id { get; set; }
        public string Login { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string Family { get; set; } = string.Empty;
        public string Surname { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string Password { get; set; } = string.Empty;
        public string Phone { get; set; } = string.Empty;
        public DateTime? Date { get; set; }
        public Adress? Adress { get; set; }
        public Guid Avatar { get; set; }   
        public Tokens? Tokens { get; set; }
        public string? Role { get; set; }
    }
}

File: e-commerce-frontend/src/app/models/account.model.ts
import { IAccountData } from "./accountData.model";
import { IAdress } from "./adress.model";
import { ITokens } from "./tokens.model";

export interface IAccount {
    id?: string, 
    login?: string,
    password?: string;
    email?: string, 
    name?: string,
    family?: string,
    surname?: string,
    date?: Date,
    phone?: string,
    adress?: IAdress,
    avatar?: string,
    tokens?: ITokens,
    role?: Role,
    accountDataModel?: IAccountData
}

export enum Role {
    admin = 'admin',
    user = 'user'
}